spras/interval.py

```diff
@@ -0,0 +1,234 @@
+"""
+Utilities for defining inequality intervals (e.g. l < x <= u)
+
+For graph heuristics, we allow inequality intervals of the form (num) < (id)?. For example,
+we can say "1500 <" for "1500 < x", or "1000 < x < 2000", etc.
+
+[If there is ever a library that does this, we should replace this code with that library.]
```

Comment:
I didn't review this file yet. I looked for any existing libraries:
- https://boolean-parser.readthedocs.io/en/latest/intro.html is the best match. I'm still reading to confirm that it avoid `eval`
- SymPy and https://docs.sympy.org/latest/modules/parsing.html looks very powerful but uses `eval`
- https://github.com/AaryamanBhute/OpenExpressions may be able to do this. It's not as clear how to use it.